journalapp/views.py

```diff
@@ -13,4 +43,35 @@ def home(request):
 def detail(request):
     pkey = request.matchdict.get("pkey")
     one_post = query_post(pkey)
+    if one_post is None:
+        raise HTTPNotFound
     return dict(post=one_post)
+
+
+@view_config(route_name='new', renderer='templates/blog_new.jinja2')
+def new(request):
+    form = NewBlogEntryForm(request.POST or NoVars())
+    if request.method == "POST":
+        new_pkey = new_entry(form.title.data, form.text.data)
```

Comment:
What if your form data is not valid?  Do you want to go ahead and make a new entry anyway?